SimG4CMS/Calo/src/HFNoseSD.cc

```diff
@@ -47,6 +47,9 @@ HFNoseSD::HFNoseSD(const std::string& name, const DDCompactView & cpv,
   waferRot_        = m_HFN.getParameter<bool>("RotatedWafer");
   angles_          = m_HFN.getUntrackedParameter<std::vector<double>>("WaferAngles");
 
+  nameX_ = ((name.find("HFNoseHits")!=std::string::npos) ? 
+	    "HGCalHFNoseSensitive" : "HFNoseSensitive");
+
```

Comment:
@bsunanda , I may be do not understand something but why sensitive detector name has "HFNoseHits" name? name is copied to SD name, which is accessible via GetName() top method. If we recognise DetID via name, this name can be used. If not, I would instead use enumeration and not strings nameX_.

Reply:
name or GetName() returns the name of the hit collection and not the name of SD.

Reply:
This has been the convention in CMS from the beginning

Reply:
@bsunanda , from what I observed in the code by fact there was no difference between these two.